src/ray/gcs/store_client/redis_store_client.cc

```diff
@@ -244,16 +218,115 @@ Status RedisStoreClient::AsyncMultiGet(
   RAY_CHECK(callback);
   if (keys.empty()) {
     callback({});
+    return Status::OK();
   }
   std::vector<std::string> true_keys;
   for (auto &key : keys) {
     true_keys.push_back(GenRedisKey(external_storage_namespace_, table_name, key));
   }
-  RAY_CHECK_OK(MGetValues(
-      redis_client_, external_storage_namespace_, table_name, true_keys, callback));
+  MGetValues(table_name, true_keys, callback);
   return Status::OK();
 }
 
+void RedisStoreClient::ProcessNext(std::string key) {
+  std::function<void()> op;
+  {
+    absl::MutexLock lock(&mu_);
+    auto op_iter = redis_ops_.find(key);
+    RAY_CHECK(op_iter != redis_ops_.end()) << " redis ops stats failed for key: " << key;
+    auto &queue = op_iter->second;
+    // All write operations have been consumed, remove this entry
+    if (queue.empty()) {
+      redis_ops_.erase(op_iter);
+    } else {
+      op = std::move(queue.front());
+    }
+  }
+  if (op) {
+    op();
+  }
+}
+
+void RedisStoreClient::SendRedisCmd(std::vector<std::string> keys,
```

Comment:
so will it split a multi commands (like multi-get or multi-set) into multiple requests?

Reply:
It won't. It's still one command and it'll be executed only once. For multi-keys, the queue is still being pushed for each key, but that's noop and only to indicate there is an in-flight request.